lib/controllers/responses.js

```diff
@@ -429,53 +388,100 @@ exports.patch = function patch(req, res) {
 
 
 exports.del = function del(req, res) {
-  Response.findOne({
+  Response.findOneAndUpdateAsync({
     'properties.survey': req.params.surveyId,
     'entries._id': req.params.responseId
-  })
-  .exec(function (error, doc) {
-    if (util.handleError(error, res)) { return; }
-    if (doc === null) {
+  }, {
+    $pull: {
+      entries: {
+        _id: req.params.responseId
+      }
+    }
+  }, {
+    new: false
```

Comment:
`findAndModify` can return the pre-modification doc or the post-modification doc, to support situations like this, where we remove something from the doc but want to have that the subdoc that we removed. Mongoose defaults to returning the post-modification doc, which makes sense if you're incrementing a counter and want the current value afterwards. But we want the old doc. The `new` option controls that behavior and defaults to `true`.